services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java

```diff
@@ -0,0 +1,344 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management.tenant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.hono.deviceregistry.util.DeviceRegistryUtils;
+import org.eclipse.hono.service.management.Filter;
+import org.eclipse.hono.service.management.SearchResult;
+import org.eclipse.hono.service.management.Sort;
+import org.eclipse.hono.util.Adapter;
+import org.junit.jupiter.api.Test;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Future;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * A suite of tests for verifying implementations of the Tenant management's 
+ * search tenants operation.
+ * <p>
+ * Concrete subclasses need to provide the service implementations under test
+ * by means of implementing the {@link #getTenantManagementService()} method.
+ * Also the subclasses should clean up any fixture in the database that has
+ * been created by individual test cases.
+ */
+public interface AbstractTenantManagementSearchTenantsTest {
```

Comment:
shouldn't this be implemented by some test class?

Reply:
yes. I missed to include that  class in the last commit. Now I have included.